lib/client/api.go

```diff
@@ -705,31 +832,120 @@ func ReadProfileStatus(profileDir string, profileName string) (*ProfileStatus, e
 	}
 
 	return &ProfileStatus{
-		Name: profileName,
-		Dir:  profileDir,
+		Name: opts.ProfileName,
+		Dir:  opts.ProfileDir,
 		ProxyURL: url.URL{
 			Scheme: "https",
-			Host:   profile.WebProxyAddr,
+			Host:   opts.WebProxyAddr,
 		},
-		Username:       profile.Username,
+		Username:       opts.Username,
 		Logins:         sshCert.ValidPrincipals,
 		ValidUntil:     validUntil,
 		Extensions:     extensions,
 		Roles:          roles,
-		Cluster:        profile.SiteName,
+		Cluster:        opts.SiteName,
 		Traits:         traits,
 		ActiveRequests: activeRequests,
-		KubeEnabled:    profile.KubeProxyAddr != "",
+		KubeEnabled:    opts.KubeProxyAddr != "",
 		KubeUsers:      tlsID.KubernetesUsers,
 		KubeGroups:     tlsID.KubernetesGroups,
 		Databases:      databases,
 		Apps:           apps,
 		AWSRolesARNs:   tlsID.AWSRoleARNs,
+		IsVirtual:      opts.IsVirtual,
 	}, nil
 }
 
+// ReadProfileFromIdentity creates a "fake" profile from only an identity file,
+// allowing the various profile-using subcommands to use identity files as if
+// they were profiles. It will set the `username` and `siteName` fields of
+// the profileOptions to certificate-provided values if they are unset.
+func ReadProfileFromIdentity(key *Key, opts ProfileOptions) (*ProfileStatus, error) {
+	// Note: these profile options are largely derived from tsh's makeClient()
+	if opts.Username == "" {
+		username, err := key.CertUsername()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		opts.Username = username
+	}
+
+	if opts.SiteName == "" {
+		rootCluster, err := key.RootClusterName()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		opts.SiteName = rootCluster
+	}
+
+	opts.IsVirtual = true
+
+	return profileFromKey(key, opts)
+}
+
+// ReadProfileStatus reads in the profile as well as the associated certificate
+// and returns a *ProfileStatus which can be used to print the status of the
+// profile.
+func ReadProfileStatus(profileDir string, profileName string) (*ProfileStatus, error) {
+	var err error
```

Comment:
Can you remove this?

Reply:
Assuming you mean the `err`, yup, will do.